Content.Server/Atmos/EntitySystems/FlammableSystem.cs

```diff
@@ -433,20 +433,43 @@ public override void Update(float frameTime)
 
                 if (flammable.FireStacks > 0)
                 {
-                    var air = _atmosphereSystem.GetContainingMixture(uid);
+                    var air = _atmosphereSystem.GetContainingMixture(uid); // we dont excite the atmos here, and let AtmosExposedComponent handle if it is necessary.
 
-                    // If we're in an oxygenless environment, put the fire out.
-                    if (air == null || air.GetMoles(Gas.Oxygen) < 1f)
+                    if (flammable.FuelGasMix != null)
                     {
-                        Extinguish(uid, flammable);
-                        continue;
+                        // If we're in no atmos extinguish.
+                        if (air == null)
+                        {
+                            Extinguish(uid, flammable);
+                            continue;
+                        }
+
+                        // If our local atmos doesnt have enough of the required gases extinguish.
+                        if (!flammable.FuelGasMix.LEQMoles(air))
+                        {
+                            Extinguish(uid, flammable);
+                            continue;
+                        }
                     }
 
                     var source = EnsureComp<IgnitionSourceComponent>(uid);
                     _ignitionSourceSystem.SetIgnited((uid, source));
 
                     if (TryComp(uid, out TemperatureComponent? temp))
-                        _temperatureSystem.ChangeHeat(uid, 12500 * flammable.FireStacks, false, temp);
+                    {
+                        // get hot
+                        EnsureComp<AtmosExposedComponent>(uid); // required for the entity to ever cool down.
+                        if (!_physicsQuery.TryComp(uid, out var physics))
+                            continue;
+
+                        var maxDeltaT = flammable.PeakFlameTemperature - temp.CurrentTemperature;
+                        if (maxDeltaT > 0) // we dont want to burn and get colder.
+                        {
+
+                            var tempBump = MathF.Sqrt(maxDeltaT * temp.SpecificHeat * physics.Mass) * flammable.FireStacks; //monotonic function, slows as it reaches maxT
+                            _temperatureSystem.ChangeHeat(uid, tempBump, false, temp);
```

Comment:
This is a really roundabout way to do `_temperatureSystem.ForceChangeTemperature`.

I still don't think this is necessarily the right way to do this, because it does not respect energy conservation in terms of energy supplied by the candle.

Reply:
I looked into this a bit more, if we do this (which is something like was originally there):
![image](https://github.com/user-attachments/assets/1e6d7b66-5a52-43f3-8396-a23c7a11e5ed)
It *does* eventually equalize like we would expect, and so we can output a constant/* amount of energy while burning into the room, and as long as something is burning the room will continue to get arbitrarily hotter.
However, as it turns out, most things contain a simply ludicrous amount of energy. irl we're talking like 15,000,000j  (15megajoules) per kg of wood. In irl this doesnt turn your fireplace into a nuclear bomb because most of the heat is lost to the sky, but a spacestation is an enclosed box, so it conserves all the energy. This would mean if we used *real* values almost any fire is immediately ultra-deadly.

Understanding that, I think we have two alternatives. 
- We can go with a constant joule output, and just make up joule quantities for entities to make fires not immediately awful, but rooms will continue to heat if you burn more stuff.
- We can do some sort of asymptoting function and have the room reach a maximum temperature based on whatever the burned entity is.

Both solutions seem fine to me, I do think the asymtote would be easier to maintain and harder to abuse.

Reply:
Yeah, so superconduction is the thing that makes the station not an ideal enclosed box. The built-in "asymptoting function"

Reply:
Or I guess we can do a hybrid, and have it asymptote with a smaller minimum joule emission.

Reply:
Well hold on. That might be total joules contained in the entity. So then, ~3hr burn time, we would get something like 1,400j/s